dear_petition/petition/models.py

```diff
@@ -380,7 +381,7 @@ def get_all_offense_records(self, include_annotations=True, filter_active=False)
 
         return qs
 
-    @property
+    @cached_property
```

Comment:
Wouldn't this result in a stale document being returned after we call `recalculate_petitions()`?